src/Confluent.SchemaRegistry.Serdes.Avro/SpecificSerializerImpl.cs

```diff
@@ -208,13 +211,23 @@ public async Task<byte[]> Serialize(string topic, T data, bool isKey)
 
                     if (!currentSchemaData.SubjectsRegistered.Contains(subject))
                     {
-                        // first usage: register/get schema to check compatibility
-                        currentSchemaData.WriterSchemaId = autoRegisterSchema
-                            ? await schemaRegistryClient
-                                .RegisterSchemaAsync(subject, currentSchemaData.WriterSchemaString)
-                                .ConfigureAwait(continueOnCapturedContext: false)
-                            : await schemaRegistryClient.GetSchemaIdAsync(subject, currentSchemaData.WriterSchemaString)
+                        if (useLatestVersion)
+                        {
+                            var latestSchema = await schemaRegistryClient.GetLatestSchemaAsync(subject)
                                 .ConfigureAwait(continueOnCapturedContext: false);
+                            currentSchemaData.WriterSchemaId = latestSchema.Id;
```

Comment:
if `use.latest.schema` is `true`, then the schema must exist (will not be auto-registered). however this is at odds with the default for `auto.register.schema`, which is `true`. same in the generic case.

Reply:
I can add a check that if `use.latest.version` is `true`, then `auto.register.schemas` must be `false`

Reply:
that works

Reply:
can you also update the Python PR?

Reply:
Yup, updated the Python PR as well